application/forms/Config/FieldsConfigForm.php

```diff
@@ -0,0 +1,195 @@
+
+            if ($this->getValue('type') !== 'array') {
+                $fieldsAssociation['other'] = 'Others';
+            }
+
+            $this->addElement('select', 'associated', [
+                'label'  => 'Associated Icinga Object Property',
+                'class'  => 'autosubmit',
+                'multiOptions' => $this->optionalEnum($fieldsAssociation),
+                'required' => true,
+            ]);
+
+            if ($this->fieldId !== null && ! $this->hasBeenSent()) {
+                if (preg_match('/\${([^}\s]+)}/', $this->fieldValue, $matches)) {
+                    if (preg_match('/^(?:host|service)\./', $matches[1])) {
+                        $this->getElement('associated')->setValue('customvar');
+                    } elseif ($matches[1] === 'hostgroup' || $matches[1] === 'servicegroup') {
+                        $this->getElement('associated')->setValue($matches[1]);
+                    } else {
+                        $this->getElement('associated')->setValue('other');
+                    }
+                } else {
+                    $this->getElement('associated')->setValue('other');
+                }
+            }
+
+            if (($this->hasBeenSent() || $this->fieldId !== null)) {
+                if ($this->getValue('associated') === 'customvar') {
+                    $this->addElement('text', $this->getValue('fields') . '_cv', [
+                        'label'       => $this->translate('Custom Variable'),
+                        'required'    => true,
+                        'description' => $this->translate(
+                            'Enter appropriate custom variable. For example host.vars.customvar or'
+                            . ' service.vars.customvar.'
+                        ),
+                    ]);
+                }
+
+                if ($this->getValue('associated') === 'other') {
+                    $this->addElement('text', $this->getValue('fields') . '_value', [
+                        'label'       => $this->translate('Field Value'),
+                        'required'    => true,
+                        'description' => $this->translate(
+                            'Enter the value for the field.'
+                        ),
+                    ]);
+                }
+
+                if (! $this->hasBeenSent()
+                    && $this->fieldId !== null
+                ) {
+                    if ($this->getValue('associated') === 'customvar') {
+                        $this->getElement($this->fieldId . '_cv')->setValue($matches[1]);
+                    }
+
+                    if ($this->getValue('associated') === 'other') {
+                        $this->getElement($this->fieldId . '_value')->setValue($this->fieldValue);
+                    }
+                }
+            }
+        }
+
+        $this->addElement('submit', 'submit', [
+            'label' => 'Add Field',
```

Comment:
Please make it changeable. You are using the same button for `add` and `edit` action.

Reply:
Please use `Save changes` as label for `edit` action.